nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -477,17 +497,31 @@ in {
     };
 
     systemd.services.mastodon-init-db = lib.mkIf cfg.automaticMigrations {
-      script = ''
-        if [ `psql ${cfg.database.name} -c \
+      script = lib.optionalString (!databaseActuallyCreateLocally) ''
+        umask 077
+
+        export PGPASSFILE
+        PGPASSFILE=$(mktemp)
+        cat > $PGPASSFILE <<EOF
+        ${cfg.database.host}:${toString cfg.database.port}:${cfg.database.name}:${cfg.database.user}:$(cat ${cfg.database.passwordFile})
+        EOF
+
+        if [ $(psql --host ${cfg.database.host} --user ${cfg.database.user} ${cfg.database.name} -c \
```

Comment:
I think `--host` and `--user` should be set via the PGHOST and PGUSER environment variables, which should be defined in `systemd.services.mastodon-init-db.environment`. That way we don't have to generate that much bash code in Nix.